cairo/ethereum/cancun/state.cairo

```diff
@@ -181,3 +188,106 @@ func get_storage{poseidon_ptr: PoseidonBuiltin*, state: State}(
 
     return value;
 }
+
+func set_storage{poseidon_ptr: PoseidonBuiltin*, state: State}(
+    address: Address, key: Bytes32, value: U256
+) {
+    alloc_locals;
+    let fp_and_pc = get_fp_and_pc();
+    local __fp__: felt* = fp_and_pc.fp_val;
+
+    // Assert that the account exists
+    let account = get_account_optional(address);
+    if (cast(account.value, felt) == 0) {
+        with_attr error_message("Cannot set storage on non-existent account") {
+            assert 0 = 1;
+        }
+    }
+
+    let storage_tries_dict_ptr = cast(state.value._storage_tries.value.dict_ptr, DictAccess*);
+    let (storage_trie_pointer) = hashdict_read{
+        poseidon_ptr=poseidon_ptr, dict_ptr=storage_tries_dict_ptr
+    }(1, &address.value);
+
+    // If the storage trie is not found, create an empty one and store it in the state
+    if (cast(storage_trie_pointer, felt) == 0) {
+        let (new_mapping_dict_ptr) = dict_new();
+        tempvar _storage_trie = new TrieBytes32U256Struct(
+            secured=bool(1),
+            default=U256(new U256Struct(0, 0)),
+            _data=MappingBytes32U256(
+                new MappingBytes32U256Struct(
+                    dict_ptr_start=cast(new_mapping_dict_ptr, Bytes32U256DictAccess*),
+                    dict_ptr=cast(new_mapping_dict_ptr, Bytes32U256DictAccess*),
+                    original_mapping=cast(0, MappingBytes32U256Struct*),
+                ),
+            ),
+        );
+
+        let new_storage_trie_ptr = cast(_storage_trie, felt);
+        hashdict_write{poseidon_ptr=poseidon_ptr, dict_ptr=storage_tries_dict_ptr}(
+            1, &address.value, new_storage_trie_ptr
+        );
```

Comment:
I personnally prefer 

```suggestion
    with poseidon_ptr, dict_ptr{
        hashdict_write(
            1, &address.value, new_storage_trie_ptr
        );
    }
```

 However it allows you to not define dict_ptr=xxx explicitly. Which is also nice. Perhaps we should stick to one style only.

Reply:
i prefer the following:
- if it's only one line, then I use implicit argument in the function call
- if i need multiple lines, then using `with` is nice